Modules/Core/Transform/include/itkTransform.hxx

```diff
@@ -470,6 +471,58 @@ Transform<TParametersValueType, NInputDimensions, NOutputDimensions>::TransformS
   return outputTensor;
 }
 
+template <typename TParametersValueType, unsigned int NInputDimensions, unsigned int NOutputDimensions>
+template <typename TImagePointer>
+void
+Transform<TParametersValueType, NInputDimensions, NOutputDimensions>::ApplyToImageMetadata(TImagePointer && image)
+{
+  using NonReferenceImagePointer = typename std::remove_reference<TImagePointer>::type;
+  static_assert(std::is_pointer<NonReferenceImagePointer>::value ||
+                  mpl::IsSmartPointer<NonReferenceImagePointer>::Value,
+                "ApplyToImageMetadata requires a raw pointer or SmartPointer.");
+
+  using ImageType = typename std::remove_const<typename std::remove_reference<decltype(*image)>::type>::type;
+
+  static_assert(Self::InputSpaceDimension == ImageType::ImageDimension,
+                "Image and transform must have the same spatial dimension");
+
+  if (!this->IsLinear())
+  {
+    itkWarningMacro(<< "ApplyToImageMetadata was invoked with non-linear transform of type: " << this->GetNameOfClass()
+                    << ". This might produce unexpected result.");
```

Comment:
"This might produce unexpected result." : "This might produce unexpected results." or "This might produce an unexpected result."